bindings/go/reader.go

```diff
@@ -27,6 +27,34 @@ import (
 	"github.com/jupiterrider/ffi"
 )
 
+type readOption struct {
+	Range *Range
+}
+
+// withReadOption is a functional option type for configuring read options.
+type withReadOption func(*readOption)
+
+// ReadFrom creates a WithReadOption that specifies the starting position for reading.
+func ReadFrom(value uint64) withReadOption {
+	return func(o *readOption) {
+		o.Range = rangeFrom(value)
+	}
+}
+
+// ReadUntil creates a WithReadOption that specifies the ending position for reading.
+func ReadUntil(value uint64) withReadOption {
+	return func(o *readOption) {
+		o.Range = rangeUntil(value)
+	}
+}
+
+// ReadBetween creates a WithReadOption that specifies a range for reading.
+func ReadBetween(start, end uint64) withReadOption {
```

Comment:
I'm guessing most golang users don't familiar with API like this.

https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#Client.GetObject